Xamarin.Essentials/Types/FileProvider.android.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using Android.App;
+using Android.Content;
+using Android.OS;
+using AndroidEnvironment = Android.OS.Environment;
+
+namespace Xamarin.Essentials
+{
+    [ContentProvider(
+        new[] { "${applicationId}.fileProvider" },
+        Name = "xamarin.essentials.fileProvider",
+        Exported = false,
+        GrantUriPermissions = true)]
+    [MetaData(
+        "android.support.FILE_PROVIDER_PATHS",
+        Resource = "@xml/xamarin_essentials_fileprovider_file_paths")]
+    public class FileProvider : global::Android.Support.V4.Content.FileProvider
+    {
+        internal static bool AlwaysFailExternalMediaAccess { get; set; } = false;
+
+        public static FileProviderLocation TemporaryLocation { get; set; }
+
+        internal static Java.IO.File GetTemporaryDirectory()
+        {
+            var root = GetTemporaryRootDirectory();
+            var dir = new Java.IO.File(root, "2203693cc04e0be7f4f024d5f9499e13");
```

Comment:
We always want to work in our folder so that we don't fill up the root of the cache or external storage with hundreds of requests. Also, we might later add a feature to remove all our own temporary files.